packages/features/flags/server/utils.ts

```diff
@@ -12,3 +12,59 @@ export async function getFeatureFlagMap(prisma: PrismaClient) {
     return acc;
   }, {} as Partial<AppFlags>);
 }
+
+interface CacheEntry {
+  value: boolean; // adapt to other supported value types in the future
+  expiry: number;
+}
+
+interface CacheOptions {
+  ttl: number; // time in ms
+}
+
+const featureFlagCache = new Map<keyof AppFlags, CacheEntry>();
+
+const isExpired = (entry: CacheEntry): boolean => {
+  return Date.now() > entry.expiry;
+};
+
+export const getFeatureFlag = async (
+  prisma: PrismaClient,
+  slug: keyof AppFlags,
+  options: CacheOptions = { ttl: 5 * 60 * 1000 }
+): Promise<boolean> => {
+  // pre-compute all app flags, each one will independelty reload it's own state after expiry.
+
+  if (featureFlagCache.size === 0) {
+    const flags = await prisma.feature.findMany({ orderBy: { slug: "asc" } });
+    flags.forEach((flag) => {
+      featureFlagCache.set(flag.slug as keyof AppFlags, {
+        value: flag.enabled,
+        expiry: Date.now() + options.ttl,
+      });
+    });
+  }
+
+  const cacheEntry = featureFlagCache.get(slug);
+
+  // Check if the flag is in the cache and not expired
+  if (cacheEntry && !isExpired(cacheEntry)) {
+    return cacheEntry.value;
+  }
+
+  // Fetch from the database if not in cache or cache is expired
```

Comment:
Nit: This comment is not needed because the line of code it's describing is well-written and self-explanatory based on the context and code directly above